content/en/blog/2023/creating-k8s-cluster-with-runtime-observability.md

```diff
@@ -0,0 +1,474 @@
+
+![Jaeger screen with services dropdown open showing apiserver, containerd and kubelet services as options](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855324762_kk25pv.png)
+
+Choose `apiserver` and click on **"Find Traces”.** Here you see traces from the
+Kubernetes control plane:
+
+![Jaeger screen showing a list of spans found for apiserver](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855412765_xe7rnr.png)
+
+Let’s run a sample command against Kubernetes with `kubectl`, like running an
+echo:
+
+```bash
+kubectl run -it --rm --image=alpine echo-command -- echo Hi
+
+# Output
+# If you don't see a command prompt, try pressing enter.
+# warning: couldn't attach to pod/echo-command, falling back to streaming logs: unable to upgrade connection: container echo-command not found in pod echo-command_default
+# Hi
+# pod "echo-command" deleted
+```
+
+And now, open Jaeger again, choose the `kubelet` service, operation `syncPod`,
+and add the tag `k8s.pod=default/echo-command`, you should be able to see spans
+related to this pod:
+
+![Jaeger screen showing a list of spans found for the syncPod operation on kubelet service](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855456007_flekwn.png)
+
+Expanding one trace, you’ll see the operations that created this pod:
+
+![Jaeger screen showing a single syncPod expanded](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855488540_mvy0ga.png)
+
+### Conclusion
+
+Even in beta, both traces for
+[kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2831-kubelet-tracing)
+and
+[apiserver](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/647-apiserver-tracing)
+can help a developer understand what’s happening under the hood in Kubernetes
+and start debugging issues.
+
+This will be helpful for developers that create custom tasks, like
+[Kubernetes Operators](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+that update internal resources to add more functionalities to Kubernetes.
+
+[Tracetest](https://tracetest.io/) is interested in seeing how
+applying trace-based testing techniques to traces from the core Kubernetes
+engine can be leveraged to verify expected functionality. With the current level
+of observability being exposed by Kubernetes, we feel it is a bit early to work
+to enable the execution of `kubectl` commands as one of the supported triggers
+for a trace-based test. We did, however, want to publish our findings in order
+to help others interested in seeing the current state of distributed tracing in
+the Kubernetes engine.
```

Comment:
```suggestion
```
I don't think this section is relevant for the OpenTelemetry blog. Can we remove it?

Reply:
Sure, we can remove it! But, I would love to praise my colleagues who helped me research and implement this use case. How about replacing it with something like this:

```
As a team focused on building an open source tool in the observability space,
the opportunity to help the overall OpenTelemetry community was important to us.
That’s why we were researching finding new ways of collecting traces from the
core Kubernetes engine. With the current level of observability being exposed by
Kubernetes we wanted to publish our findings in order to help others interested
in seeing the current state of distributed tracing in the Kubernetes engine.
Daniel Dias and Sebastian Choren are working on Tracetest, an open-source tool
that allows you to develop and test your distributed system with OpenTelemetry.
It works with any OTel compatible system and enables trace–based tests to be
created. Check it out at https://github.com/kubeshop/tracetest.
```

Do you think it makes sense? If not, the is no problem, I can remove this paragraph.